duckduckgo/macos-browser
# Add email feature

Task/Issue URL: https://app.asana.com/0/72649045549333/1199706899452795/f
Tech design: How to share email code between macOS and iOS: https://app.asana.com/0/481882893211075/1199925934231710/f
CC: @tomasstrba 

**Description**:
This PR integrates the email beta features into the app:

Sign in
Sign out
Autofill
Alias generation

It's similar to the feature on iOS (https://dub.duckduckgo.com/duckduckgo/ios-email/pull/1), except without the copy to clipboard option

We're still waiting on deciding an icon from  "..." menu

**Steps to test this PR**:
1. Go to anywhere with an email sign up form, dax should not be visible on the input
2. From the "..." menu, select "Turn on email protection". This should open a new tab with the email signup/login page.
- Then login.
- Now go to anywhere with an email sign up form. Dax should be visible
- Tap dax, the email tooltip should show up.
- Tap outside the tooltip, it should disappear
- Repeat above, but press cancel, the tooltip should disappear
- Tap dax again, but this time select "Use Address". The email field should be populated with the alias shown in the tooltip.
- Tap dax again, the tooltip should show with a new alias. Tap "Use address" again, and the email filed should be populated with the new tooltip.
3. From the "..." menu, select "Turn off email protection". This should log you out. Now go to anywhere with an email sign up form, dax should not be visible on the input
4. This part is particularly **important** , test no other UserScripts have broken or changed behaviour now that they've been migrated over to use the BrowserServicesKit UserScript 


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
